### PR TITLE
#4950 re-enable viewing remote images in signed/encrypted messages

### LINKED ISSUE #4950: re-enable viewing remote images in signed/encrypted messages
... (edited - see below) ...

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR re-enables viewing remote images in signed/encrypted messages

close #4950 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

chrome-capture-2023-1-21

### CODE REVIEW

**Review by sosnovsky — COMMENTED** [review]
Looks good, just needs loader update

File: extension/js/common/platform/xss.ts [hunk 1]
```diff
@@ -151,7 +151,9 @@ export class Xss {
             a.setAttribute('data-test', 'show-inline-image');
             Xss.replaceElementDANGEROUSLY(img, a.outerHTML); // xss-safe-value - "a" was build using dom node api
           } else {
-            Xss.replaceElementDANGEROUSLY(img, `<span>[Remote images are blocked due to security]</span>`); // xss-safe-value
+            img.setAttribute('data-src', img.getAttribute('src') ?? '');
+            img.classList.add('replace_to_base64_image');
+            img.setAttribute('src', '/img/loading.gif');
```
Comment (sosnovsky) [inline extension/js/common/platform/xss.ts:156]: let's use our standard loader `UI.spinner` - it's more lightweight as uses `svg`

**Review by sosnovsky — COMMENTED** [review]
Looks good, just needs loader update

File: extension/js/common/platform/xss.ts (hunk 1 above)
Comment (ioanmo226) [reply extension/js/common/platform/xss.ts:156]: Ok, Sounds good

**Review by sosnovsky — COMMENTED** [review]
I found issue with this new implementation - images disappear when you try to forward message or reply to it

**Comment by tomholub** [comment]
> I found issue with this new implementation - images disappear when you try to forward message or reply to it

That won't be very easy to fix, since sent messages use text/plain and not text/html. Let's file it for later?

**Comment by sosnovsky** [comment]
Maybe it'll be easier to fix as we're converting remote images to base64. Or we can at least leave image urls there, as currently it's just empty space instead of images.

**Comment by tomholub** [comment]
> Maybe it'll be easier to fix as we're converting remote images to base64. Or we can at least leave image urls there, as currently it's just empty space instead of images.

We can indeed start with image urls then - that's a good idea.

For the base64 images, you'd have to attach them as files, which would make sense for forwarding?

**Review by ioanmo226 — COMMENTED** [review]
@tomholub , @sosnovsky 
In `htmlSanitizeAndStripAllTags` function, we convert html to platin text + remove img tags.
That's why images are removed when forwarding message.
So do we have to keep image tags in `htmlSanitizeAndStripAllTags` function?

https://github.com/FlowCrypt/flowcrypt-browser/blob/6249a4bcd5ebc6e22e7a61ea33554215a583dab3/extension/js/common/platform/xss.ts#L187

**Comment by tomholub** [comment]
no, just replace remote images with urls to the image within the resulting plain text

**Comment by ioanmo226** [comment]
You mean forwarded message should be `Test test.com/img message` when original message is `Test  message`

**Comment by tomholub** [comment]
the first one. it's plain text, not html, no point putting html tags in there

**Comment by ioanmo226** [comment]
ok

**Review by sosnovsky — APPROVED** [review]
Well done 👍

**Pull request merged**